PERFORMANCE.md

```diff
@@ -0,0 +1,98 @@
+# Performance
+
+It is important to be conscious of performance in hot code paths. In such code paths there should be 3 goals:
+
+- Be as fast as possible
+- Be memory efficient
+- Be as stable as possible
+
+This guide is meant to document known performance issues so that the contributors are aware of them.
```

Comment:
@chadhietala this is very interesting. I like the fact that this document will change over time as well. Seems like a good resource to promote ways to keep performance considerations in mind. 👍